packages/space-opera/src/components/mobile_view/mobile_view.ts

```diff
@@ -52,82 +56,69 @@ export class MobileView extends ConnectedLitElement {
   @internalProperty() snippetPipeUrl = `${this.base}-state-${this.pipeId}`;
   @internalProperty() updatesPipeUrl = `${this.base}-updates-${this.pipeId}`;
   @internalProperty() mobilePingUrl = `${this.base}-ping-${this.pipeId}`;
+  @internalProperty() envPipeUrl = `${this.base}-env-${this.pipeId}`;
 
   stateChanged(state: State) {
-    this.gltfUrl = getGltfUrl(state);
     this.config = getConfig(state);
     this.arConfig = getArConfig(state);
     this.hotspots = getHotspots(state);
     this.camera = getCamera(state);
-    this.gltf = getGltfModel(state);
-  }
-
-  getSrcPipeUrl(srcType: string): string {
-    return `https://ppng.io/modelviewereditor-srcs-${srcType}-${this.pipeId}`;
   }
 
-  // TODO: https://javascript.info/fetch-progress
-  async waitForModel() {
-    await fetch(this.getSrcPipeUrl('gltf'))
-        .then(response => response.blob())
-        .then(blob => {
-          const modelUrl = URL.createObjectURL(blob);
-          reduxStore.dispatch(dispatchGltfUrl(modelUrl));
-          reduxStore.dispatch(dispatchSetHotspots([]));
-        })
-        .catch((error) => {
-          console.error('Error:', error);
-        });
+  setNewModelSrc(id: number) {
```

Comment:
instead of fetching the model, then setting the model-viewer src to that new url, we are letting model-viewer handle the fetching of the model. 

this is to ensure consistency between the scene-viewer POST and the model-viewer src.

this also allows us to utilize model-viewer's progress bar